tests/foreman/ui/test_errata.py

```diff
@@ -163,37 +145,132 @@ def vm(module_repos_collection_with_setup, rhel7_contenthost, target_sat):
     return rhel7_contenthost
 
 
+@pytest.fixture
+def registered_contenthost(
+    rhel_contenthost,
+    module_org,
+    module_lce,
+    module_cv,
+    module_target_sat,
+    request,
+    repos=[CUSTOM_REPO_URL],
+):
+    """RHEL ContentHost registered in satellite,
+    Using SCA and global registration.
+
+    :param repos: list of upstream URLs for custom repositories,
+        default to CUSTOM_REPO_URL
+    """
+    activation_key = module_target_sat.api.ActivationKey(
+        organization=module_org,
+        environment=module_lce,
+    ).create()
+
+    custom_products = []
+    custom_repos = []
+    for repo_url in repos:
+        # Publishes a new cvv, associates org, ak, cv, with custom repo:
+        custom_repo_info = module_target_sat.cli_factory.setup_org_for_a_custom_repo(
+            {
+                'url': repo_url,
+                'organization-id': module_org.id,
+                'lifecycle-environment-id': module_lce.id,
+                'activationkey-id': activation_key.id,
+                'content-view-id': module_cv.id,
+            }
+        )
+        custom_products.append(custom_repo_info['product-id'])
+        custom_repos.append(custom_repo_info['repository-id'])
+
+    # Promote newest version with all content
+    module_cv = module_cv.read()
+    module_cv.version.sort(key=lambda version: version.id)
+    module_cv.version[-1].promote(data={'environment_ids': module_lce.id})
+    module_cv = module_cv.read()
+
+    result = rhel_contenthost.register(
+        activation_keys=activation_key.name,
+        target=module_target_sat,
+        org=module_org,
+        loc=None,
+    )
+    assert result.status == 0, f'Failed to register host:\n{result.stderr}'
+    assert rhel_contenthost.subscribed
+
+    for custom_repo_id in custom_repos:
+        custom_repo = module_target_sat.api.Repository(id=custom_repo_id).read()
+        assert custom_repo
+        result = custom_repo.sync()['humanized']
+        assert (
+            len(result['errors']) == 0
+        ), f'Failed to sync custom repository [id: {custom_repo_id}]:\n{str(result["errors"])}'
+
+    yield rhel_contenthost
+
+    @request.addfinalizer
+    # Cleanup for in between parameterized runs
+    def cleanup():
```

Comment:
It looks that in this cleanup we just roll back all the changes so that we have the module-scoped fixtures (mostly `module_cv`) ready for the next RHEL contenthost (via parametrization). What is the idea behind this need? Isn't it just fine to go with function-scoped fixtures instead module-scoped?

Reply:
I could switch them to function fixtures, but may not be a simple change. 
Many of the errata tests already run using module scope fixtures, so was keeping those for consistency.

Reply:
Ok, no worries, merged already.